microcosm_eventsource/event_types.py

```diff
@@ -195,6 +203,52 @@ def all_states(cls):
                     next_states.append(new_state)
                     known_states.add(new_state)
 
+    @classmethod
+    def all_states_and_events(cls):
+        """
+        Return a generator of all allowed (state, event_type) combinations
+        Note: it can return the same state or event twice (but all uniqu)
+
+        """
+        initial_states = list(cls.initial_states())
+        initial_states_and_events = [(state, list(state)[0]) for state in initial_states]
```

Comment:
we are assuming that the first item in the state will be the same as the event?

Reply:
Good catch -it might be false